features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/ConfirmPinFragment.kt

```diff
@@ -81,34 +81,40 @@ class ConfirmPinFragment : BaseDialogFragment(), OtpSubmissionClickHandler, OtpC
             bundle.getString(PIN)?.let { pin ->
                 if (confirmedPin == pin) {
                     showProgressBar(true)
-                    viewModel.createPinResponse.observe(this, Observer {
+                    viewModel.createPinResponse.observe(this, Observer { payloadResource ->
 
-                        when (it) {
-                            is Loading -> showProgressBar(it.isLoading, getString(R.string.creating_pin))
+                        when (payloadResource) {
+                            is Loading -> showProgressBar(payloadResource.isLoading, getString(R.string.creating_pin))
                             is Success -> {
-                                activity?.let {
-                                    it.setPinCreated(true)
+                                activity?.let { fragmentActivity ->
```

Comment:
instead just do activity?.setPinCreated() or activity?.setResult, activity?.setFinish(). let creates a scoped function, so might not be useful considering the observer below has to be moved out.

Reply:
I have just added existing logic inside `.let` everywhere. If the said change is required, it was required before also.